tests/10apidoc/20presence.pl

```diff
@@ -71,23 +71,23 @@ sub matrix_set_presence_status
 
    proves => [qw( can_set_presence )],
 
-   do => sub {
+   check => sub {
       my ( $user ) = @_;
 
       matrix_set_presence_status( $user, "online",
          status_msg => $status_msg,
-      )
-   },
-
-   check => sub {
-      my ( $user ) = @_;
-
-      matrix_get_presence_status( $user )->then( sub {
-         my ( $body ) = @_;
-
-         ( $body->{status_msg} // "" ) eq $status_msg or
-            die "Incorrect status_msg";
-
-         Future->done(1);
-      });
+      )->then( sub {
+         # If presence is on a different worker it may take a while for it to
+         # propagate.
```

Comment:
Users always receive their own presence down sync: https://github.com/matrix-org/sytest/blob/0d59907cc6c9de33b8aa37c8dbfd75d9fa9c755c/tests/31sync/05presence.pl#L3

In which case, it might be better to write a `matrix_set_presence_status_synced` method.

...but also: meh, this works.